pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,46 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Steps Steps `json:"steps,omitempty" protobuf:"bytes,11,opt,name=steps"`
```

Comment:
At some point, jenkins needs to record status in a real object.  When it does that, I assume it'll be stages, with steps in them.  For this PR can you describe this in terms of terminology jenkins uses, and the pipeline plugin will use, so we can end up with a struct in Status that looks as much as possible like the destination will be?